src/controller.cpp

```diff
@@ -252,14 +261,17 @@ void controller::run(int argc, char * argv[]) {
 
 	/* First of all, let's check for options that imply silencing of the
 	 * output: import, export, command execution and, well, quiet mode */
+	char *homedir = NULL;
 	while ((c = ::getopt_long(argc, argv, getopt_str, longopts, NULL)) != -1) {
 		if (strchr("iexq", c) != NULL) {
 			silent = true;
-			break;
+		}
+		if(strchr("H", c) != NULL){
+			homedir = optarg;
 		}
 	}
 
-	setup_dirs(silent);
+	setup_dirs(homedir, silent);
 
 	/* Now that silencing's set up, let's rewind to the beginning of argv and
 	 * process the options */
```

Comment:
This one is also not true since you handled not only silencing but also the homedir. Update it, please.